### ISSUE TITLE
Add an option to disable reportUnknownArgumentType for lambdas only

### ISSUE THREAD

**Issue by Kein** [microsoft/pylance-release#699]
Consider the following:
```python
my_map = dict(numValues = {'reader': lambda x: unpack_int(x)[0]}
```

If you have `reportUnknownArgumentType` set to anything above `none` it will report this. Problem is, as far as I know, in that particular setup, you cannot annotate lambda properly with type hints. But if you disable `reportUnknownArgumentType` you will lose the ability completely. 

I don't think that's a fair trade and I think adding extra diagnostics option is lesser evil and less crumblesome than plaguing your code with `#type: ignore` all the way.

**Comment by erictraut** [comment]
You are correct that Python unfortunately doesn't provide any syntax for annotating the input parameters for a lambda, but you can provide an annotation outside of the lambda that provides sufficient context.

I can provide more details if you supply a working self-contained sample. The sample you've provided above has several errors and omissions.

**Comment by Kein** [comment]
The screenshot wasn't an example, it was intended to showcase how verbose and noisy it becomes.
Yes, I can provide an example:
```python
import struct
import binascii

b = b'\x01\x00\x00\x00'
unpack_int = struct.Struct('<I').unpack

my_map = dict(numValues = {'reader': lambda x: unpack_int(x)[0]})

for key, value in my_map.items():
	f = value['reader']
	value['value'] = f(b[0:4])

print(my_map)
```

but the thing is, **I still insist** that the feature in question would be useful to avoid needless noise. Perhaps implemented in this way as described here:
https://github.com/microsoft/pylance-release/issues/282#issuecomment-740377743
